linera-views/src/backends/rocks_db.rs

```diff
@@ -425,19 +401,30 @@ impl AdminKeyValueStore for RocksDbStoreInternal {
         namespace: &str,
         root_key: &[u8],
     ) -> Result<Self, RocksDbStoreInternalError> {
-        Self::check_namespace(namespace)?;
-        let mut path_buf = config.path_with_guard.path_buf.clone();
-        let mut path_with_guard = config.path_with_guard.clone();
-        path_buf.push(namespace);
-        path_with_guard.path_buf = path_buf;
-        let max_stream_queries = config.common_config.max_stream_queries;
-        let spawn_mode = config.spawn_mode;
-        RocksDbStoreInternal::build(path_with_guard, spawn_mode, max_stream_queries, root_key)
+        let start_key = vec![1];
+        let mut store = RocksDbStoreInternal::build(config, namespace, start_key)?;
+        let mut batch = Batch::new();
+        batch.put_key_value_bytes(root_key.to_vec(), vec![]);
+        store.write_batch(batch).await?;
+
+        let mut start_key = vec![0];
+        start_key.extend(root_key);
+        store.executor.start_key = start_key;
+        Ok(store)
     }
 
-    fn clone_with_root_key(&self, root_key: &[u8]) -> Result<Self, RocksDbStoreInternalError> {
+    async fn clone_with_root_key(
+        &self,
+        root_key: &[u8],
+    ) -> Result<Self, RocksDbStoreInternalError> {
         let mut store = self.clone();
-        store.executor.root_key = root_key.to_vec();
+        store.executor.start_key = vec![1];
```

Comment:
I feel that we should define two global constants instead of `0` and `1`.

Reply:
The notion of "key space" or "domain" comes to mind.

Reply:
Ok, except for the second where I prefer `STORE_ROOT_KEYS_PREFIX`.